lucene/src/main/java/org/hibernate/hql/lucene/internal/ClassBasedLuceneQueryResolverDelegate.java

```diff
@@ -313,7 +313,7 @@ public void propertyPathCompleted(PropertyPath path) {
 
 	@Override
 	public void registerJoinAlias(Tree alias, PropertyPath path) {
-		if ( !path.getNodes().isEmpty() && !aliasToPropertyPath.containsKey( alias ) ) {
+		if ( !path.getNodes().isEmpty() && !aliasToPropertyPath.containsKey( alias.getText() ) ) {
```

Comment:
For a typo, no worries, we're generally all happy with a separate commit.
But is this a typo?